pygmt/clib/conversion.py

```diff
@@ -100,6 +100,10 @@ def dataarray_to_matrix(grid):
                     dim
                 )
             )
+        if coord_inc == 0:
+            raise GMTInvalidInput(
+                "Grid has zero increment in the '{}' dimension.".format(dim)
```

Comment:
I think we are slowly converting format strings to f-strings, although there are still many format strings through the code.
```suggestion
                f"Grid has a zero increment in the '{dim}' dimension."
```

Reply:
done